.openshift-ci/tests/e2e-test.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+export GITROOT="$(git rev-parse --show-toplevel)"
+source "${GITROOT}/dev/env/scripts/lib.sh"
+init
+
+up.sh
+
+log "Environment up and running"
+log "Waiting for fleet-manager to complete leader election..."
+# Don't have a better way yet to wait until fleet-manager has completed the leader election.
+$KUBECTL -n "$ACSMS_NAMESPACE" logs -l application=fleet-manager -c fleet-manager -f |
+    grep --line-buffered 'Running as the leader and starting' |
+    head -1 >/dev/null || true
+sleep 1
+
```

Comment:
Probably it is due to my limited bash skills but I don't understand how this stops following logs on the first match, can you explain? I expected something like this:
```suggestion
$KUBECTL -n "$ACSMS_NAMESPACE" logs -l application=fleet-manager -c fleet-manager -f |
    grep --line-buffered -m1 'Running as the leader and starting' > /dev/null
sleep 1

```
Also why do we need the sleep afterwards?

Reply:
I didn't know about `grep -m`, thanks! That's a simplification.
Well, with `head` it also works because head will stop reading after it has retrieved the first line (which is the first match), this will then terminate the whole pipe, all the way back to `kubectl logs`.

Thoughout this PR you might here and there see a `sleep 1`. Sometimes even in the presence of a technically more correct wait condition. These sleeps are there because they seem to prevent certain races/failures while bringing up the environment. It is totally not obvious why you need them, but I have sometimes seen failures without them and they seem to fix it. Of course it would be better to get rid of them and use proper wait conditions wherever possible.
For now I wouldn't care too much about them if it works.

Reply:
I guess in this particular case the situation was that there was small time window after fleet-manager has finished its leader election during which API calls triggered by the tests would still fail.